# PATCH_WORKING_DIR: .
diff --git a/storage/storage.nim b/storage/storage.nim
index 16a2cb20..99c8add3 100644
--- a/storage/storage.nim
+++ b/storage/storage.nim
@@ -21,7 +21,8 @@ import pkg/libp2p
 import pkg/confutils
 import pkg/confutils/defs
 import pkg/stew/io2
-import pkg/datastore
+when defined(CLIENT_LITE):
+  import pkg/datastore/sql
 import pkg/stew/io2
 
 import ./node
@@ -181,7 +182,10 @@ proc new*(
     )
 
   let providersPath = config.dataDir / StorageDhtProvidersNamespace
-  let discoveryStoreRes = LevelDbDatastore.new(providersPath)
+  when defined(CLIENT_LITE):
+    let discoveryStoreRes = SQLiteDatastore.new(providersPath / "discovery.db")
+  else:
+    let discoveryStoreRes = LevelDbDatastore.new(providersPath)
   if discoveryStoreRes.isErr:
     error "Failed to initialize discovery datastore",
       path = providersPath, err = discoveryStoreRes.error.msg
@@ -200,32 +204,44 @@ proc new*(
 
     network = BlockExcNetwork.new(switch)
 
-    repoData =
-      case config.repoKind
-      of repoFS:
-        Datastore(
-          FSDatastore.new($config.dataDir, depth = 5).expect(
-            "Should create repo file data store!"
-          )
-        )
-      of repoSQLite:
-        Datastore(
-          SQLiteDatastore.new($config.dataDir).expect(
-            "Should create repo SQLite data store!"
+    when defined(CLIENT_LITE):
+      repoData = Datastore(
+        SQLiteDatastore.new($config.dataDir).expect(
+          "Should create repo SQLite data store!"
+         )
+      )
+    else:
+      repoData =
+        case config.repoKind
+        of repoFS:
+          Datastore(
+            FSDatastore.new($config.dataDir, depth = 5).expect(
+              "Should create repo file data store!"
+            )
+           )
+        of repoSQLite:
+          Datastore(
+            SQLiteDatastore.new($config.dataDir).expect(
+              "Should create repo SQLite data store!"
+            )
           )
-        )
-      of repoLevelDb:
-        Datastore(
-          LevelDbDatastore.new($config.dataDir).expect(
-            "Should create repo LevelDB data store!"
+        of repoLevelDb:
+          Datastore(
+            LevelDbDatastore.new($config.dataDir).expect(
+              "Should create repo LevelDB data store!"
+            )
           )
-        )
 
     repoStore = RepoStore.new(
       repoDs = repoData,
-      metaDs = LevelDbDatastore.new(config.dataDir / StorageMetaNamespace).expect(
-          "Should create metadata store!"
-        ),
+      when defined(CLIENT_LITE):
+        metaDs = SQLiteDatastore.new((config.dataDir / StorageMetaNamespace) & ".db").expect(
+            "Should create metadata store!"
+          )
+      else:
+        metaDs = LevelDbDatastore.new(config.dataDir / StorageMetaNamespace).expect(
+            "Should create metadata store!"
+          ),
       quotaMaxBytes = config.storageQuota,
       blockTtl = config.blockTtl,
     )
