# PATCH_WORKING_DIR: vendor/nim-datastore
diff --git a/datastore.nim b/datastore.nim
index 38bd5f6..81eb291 100644
--- a/datastore.nim
+++ b/datastore.nim
@@ -1,8 +1,12 @@
 import ./datastore/datastore
 import ./datastore/fsds
 import ./datastore/sql
-import ./datastore/leveldb
+when not defined(CLIENT_LITE):
+  import ./datastore/leveldb
 import ./datastore/mountedds
 import ./datastore/tieredds
 
-export datastore, fsds, mountedds, tieredds, sql, leveldb
+when defined(CLIENT_LITE):
+  export datastore, fsds, mountedds, tieredds, sql
+else:
+  export datastore, fsds, mountedds, tieredds, sql, leveldb
